### PR TITLE
[Consignments] Adds Artist match search to metaphysics

### PR DESCRIPTION
**Pull request by orta**
Adds support for querying Artists using Elastic Search.

screen shot 2017-05-23 at 15 42 17

### CODE REVIEW

File: test/schema/match/artist.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    beforeEach(() => {
+      gravity = sinon.stub()
+      MatchArtist.__Rewire__("gravity", gravity)
+    })
+
+    afterEach(() => {
+      MatchArtist.__ResetDependency__("gravity")
+    })
+
+    it("returns bidder ids that the user is registered in sales for", () => {
```
Comment (mzikherman) [inline test/schema/match/artist.js]: 🍝

File: test/schema/match/artist.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      gravity.withArgs("match/artists", { term: "ok" }).returns(Promise.resolve(response))
+
+      return runQuery(query).then(data => {
+        expect(data).toEqual({ match_artist: [{ birthday: "1958", id: "han-myung-ok", name: "Han Myung-Ok" }] })
+      })
+    })
+  })
+})
+
+// console.log("OK?")
```
Comment (mzikherman) [inline test/schema/match/artist.js]: 🌴

**Comment by mzikherman** [comment]
LGTM except for the minor test errata.

Re: `artist_match`, I'm definitely _intrigued_ by the more generic `search(term: 'orta', type: Artist)` (esp if `type` could take arrays of enum'ed values too), but also that's probably premature. That might be if someone needs a more nuanced search. Just for regular auto-complete for a certain type, I think the more explicit `match_...` is cool 👍

**Comment by orta** [comment]
I had this discussion with @joeyAghion in slack - turns out the generic search (scoped to artists) is different from the artist search - so putting them into one single query mixed with different modifiers would end up being confusing

Good catch on the 🍝

File: test/schema/match/artist.js (hunk 2 above)
Comment (orta) [inline test/schema/match/artist.js]: yeah, I forgot that in order to use jest mocks I needed to remove all instances of babel-rewire - see https://github.com/artsy/metaphysics/pull/635 for that stone to roll

**Comment by orta** [comment]
Updated 👍

File: schema/match/artist.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// @flow
+import type { GraphQLFieldConfig } from "graphql"
+
+import { GraphQLString, GraphQLList, GraphQLNonNull, GraphQLInt } from "graphql"
```
Comment (mzikherman) [inline schema/match/artist.js]: surprised the 👮 didnt get you for not alphabetizing these!

Comment (orta) [inline schema/match/artist.js]: Interesting,  yeah, it should be

**Pull request merged**